recharges/tasks.py

```diff
@@ -92,10 +91,45 @@ def look_up_mobile_operator(msisdn):
         return False
 
 
+class ReadyRecharge(Task):
+
+    """
+    Task to set the normalise the msisdn and attempt to set the
+    network operator based on the leading msisdn characters
+    """
+
+    def run(self, recharge_id, **kwargs):
+        l = self.get_logger(**kwargs)
+        recharge = get_recharge(recharge_id)
+
+        # Normalize the msisdn
+        recharge.msisdn = normalize_msisdn(recharge.msisdn, '27')
+        recharge.save()
+
+        # Set the network operator
+        network = slice_network_operator(recharge.msisdn)
+        if not network:
+            # If no network is found, mark the recharge unrecoverable
+            # gsvr: Perhaps we should rather use a different status here
```

Comment:
We'll add a status field that we can dump reasons for status ID in - we need that for Hotsocket response parsing too